helix-view/src/editor.rs

```diff
@@ -1469,6 +1469,17 @@ impl Editor {
             doc.restore_cursor = false;
         }
     }
+
+    /// Update read-only registers.
+    pub fn update_registers(&mut self) {
+        let doc = doc!(self);
+        if let Some(path) = doc.relative_path() {
+            self.registers
+                .write('%', vec![path.to_string_lossy().to_string()]);
+        } else {
+            self.registers.clear('%');
```

Comment:
To match kakoune we could write the scratch buffer name into the register when it doesn't have a path

Reply:
Yeah, I guess that's fine too.